src/scribe_data/cli/total.py

```diff
@@ -19,3 +19,93 @@
     * along with this program.  If not, see <https://www.gnu.org/licenses/>.
     -->
 """
+
+from SPARQLWrapper import SPARQLWrapper, JSON
+
+# Dictionary to map language names to their Wikidata Q-IDs
+language_to_qid = {
+    "english": "Q1860",
+    "french": "Q150",
+    "german": "Q188",
+    "italian": "Q652",
+    "portuguese": "Q5146",
+    "russian": "Q7737",
+    "spanish": "Q1321",
+    "swedish": "Q9027"
+}
+
+# Dictionary to map data types to their Wikidata Q-IDs
+data_type_to_qid = {
+    "nouns": "Q1084",
+    "prepositions": "Q37649",
+    "verbs": "Q24905",
+    "translations": "Q7553789"
+}
+
+def get_qid_by_input(input_str):
+    """
+    Retrieve the QID for a given language or data type input string.
+
+    Parameters
+    ----------
+    input_str : str
+        The input string representing a language or data type.
+
+    Returns
+    -------
+    str or None
+        The QID corresponding to the input string, or None if not found.
+    """
+    if input_str:
+        input_str_lower = input_str.lower()
+        if input_str_lower in language_to_qid:
```

Comment:
Putting a space before all `else` statements and maybe between the `elif` and the `return` would be good :)